lib/cpp_peripheral_states.dart

```diff
@@ -0,0 +1,445 @@
+        isFeatureSupported(Feature.variantReason)) {
+      await sendCommandToPrtipheral(join(Command.end, variantReason));
+    } else if (reason != null) {
+      await sendCommandToPrtipheral(join(Command.end, reason));
+    } else {
+      sendCommandToPrtipheral(join(Command.end, EndReason.undefined));
+    }
+
+    if (score != null && isFeatureSupported(Feature.score)) {
+      await sendCommandToPrtipheral(join(Command.score, score));
+    }
+  }
+
+  @override
+  Future<void> handleCentralUndo({
+    required String move,
+    String? lastMove,
+    String? check,
+    String? time,
+  }) async {
+    round.isMoveRejected = false;
+    sendRoundUpdateToCentral();
+
+    await sendCommandToPrtipheral(join(Command.undo, move));
+
+    if (lastMove != null && isFeatureSupported(Feature.lastMove)) {
+      await sendCommandToPrtipheral(join(Command.lastMove, lastMove));
+    }
+    if (check != null && isFeatureSupported(Feature.check)) {
+      await sendCommandToPrtipheral(join(Command.check, check));
+    }
+    if (time != null && isFeatureSupported(Feature.time)) {
+      await sendCommandToPrtipheral(join(Command.time, time));
+    }
+  }
+
+  @override
+  Future<void> handleCentralDrawOffer() async {
+    transitionTo(CentralDrawOfferState());
+    await sendCommandToPrtipheral(Command.drawOffer);
+  }
+}
+
+class RoundBeginState extends RoundState {
+  @override
+  Future<void> handlePeripheralCommand(String cmd) async {
+    if (cmd.startsWith(Command.sync)) {
+      round.isStateSynchronized = true;
+      round.isStateSetible = false;
+      round.fen = getCommandParams(cmd);
+      round.isMoveRejected = false;
+      transitionTo(RoundOngoingState());
+      sendRoundInitializedToCentral();
+      sendStateSynchronizeToCentral(true);
+    } else if (cmd.startsWith(Command.unsyncSetible)) {
+      round.isStateSynchronized = false;
+      round.isStateSetible = true;
+      round.fen = getCommandParams(cmd);
+      round.isMoveRejected = false;
+      transitionTo(RoundOngoingState());
+      sendRoundInitializedToCentral();
+      sendStateSynchronizeToCentral(false);
+    } else if (cmd.startsWith(Command.unsync)) {
+      round.isStateSynchronized = false;
+      round.isStateSetible = false;
+      round.fen = getCommandParams(cmd);
+      round.isMoveRejected = false;
+      transitionTo(RoundOngoingState());
+      sendRoundInitializedToCentral();
+      sendStateSynchronizeToCentral(false);
+    } else {
+      await super.handlePeripheralCommand(cmd);
+    }
+  }
+}
+
+class RoundOngoingState extends RoundState {
+  @override
+  Future<void> handlePeripheralCommand(String cmd) async {
+    if (cmd.startsWith(Command.move)) {
+      final move = getCommandParams(cmd);
+      round.lastMove = move;
+      round.isMoveRejected = false;
+      transitionTo(PeripheralMoveState());
+      sendMoveToCentral(move);
+    } else if (cmd.startsWith(Command.sync)) {
+      round.isStateSynchronized = true;
```

Comment:
just idea.
I have noticed that flow is always the same
round.something1 = value1
round.something2 = value2
sendRoundUpdateToCentral()

What about having some method like
round.updateWitCentralNotify(something1=value1, something2)

where updateWitCentralNotify will be some method with optional parameters, which reflect things which can be set on round?"

Reply:
Hmm, in case of:
```dart
  Future<void> handleCentralReject() async {
    round.isMoveRejected = true;
    sendRoundUpdateToCentral();
```
will be a lot of `if not` calls like:
```dart
if (isStateSynchronized) {
  round.isStateSynchronized = isStateSynchronized;
}
if (isStateSetible) {
  round.isStateSetible = isStateSetible;
}
if (fen) {
  round.fen = fen;
}
```
will be called. Don't like such solution